src/main/java/duke/tasks/Task.java

```diff
@@ -0,0 +1,68 @@
+package duke.tasks;
+
+/**
+ * Abstract base task created by Duke.
+ */
+abstract public class Task {
+    String name;
+    boolean isTaskDone = false;
+
+    /**
+     * Default constructor for a new task.
+     *
+     * @param name task name.
+     */
+    public Task(String name) {
+        this(name, false);
+    }
+
+    /**
+     * Full constructor with all fields exposed
+     *
+     * @param name task name
+     * @param isTaskDone done state
+     */
+    public Task(String name, boolean isTaskDone) {
+        this.name = name;
+        this.isTaskDone = isTaskDone;
+    }
+
+    /**
+     * Generates the message printed when this task is added.
+     *
+     * @return added: task name.
+     */
+    public String addMsg() {
```

Comment:
Perhaps change Msg to Message to make it easier to understand for those who may not recognise what Msg means